.github/workflows/publish-canary.yml

```diff
@@ -20,6 +20,10 @@ jobs:
         with:
           node-version: 12.16.3
 
+      - name: Succeed early if a canary has already been published at this commit
+        if: $(npm view @statechannels/server-wallet dist-tags.next) =~ $(git rev-parse --short HEAD)
```

Comment:
I have singled out `server-wallet` here, but *all* packages are published when in canary mode (so no need to check the others).